src/converse-chatview.js

```diff
@@ -344,11 +344,19 @@
                                "Output has been shortened."),
                             true, true);
                     }
+
+                    var mess_time;
+
+                    if(_converse.time_format === 'h')
+                        mess_time = msg_time.format('hh:mm a');
+                    else if(_converse.time_format === 'H')
+                        mess_time = msg_time.format('HH:MM');
+
                     var $msg = $(template(
                         _.extend(this.getExtraMessageTemplateAttributes(attrs), {
                             'msgid': attrs.msgid,
                             'sender': attrs.sender,
-                            'time': msg_time.format('hh:mm'),
+                            'time': mess_time,
```

Comment:
ok, so I'm making a change with, keeping three settings values then, 'HH:MM', 'hh:mm' and 'hh:mm a', which will give a 24hr time format, 12hr time format, and a 12hr time format with am/pm displayed.